src/main/webapp/scripts/place-suggestions.js

```diff
@@ -42,10 +44,110 @@ function findPlacesInCity(city, filter) {
+  )
+  const { results } = await placesResponse.json();
+  getPlaceCardInformation(results);
+}
 
+async function getPlaceCardInformation(places) {
+  let placeDetailsArr = [];
+  for(let i = 0; i < places.length; i++) {
+    const { place_id } = places[i];
+    const placeDetails = await getPlaceDetails(place_id);
+    placeDetailsArr.push(placeDetails);
+  }
 
+  renderPlaceCards(placeDetailsArr);
+}
 
+function renderPlaceCards(places) {
+  let placeCards = [];
+  for(let i = 0; i < places.length; i++) {   
+    const { phoneNumber, name, photoUrl, priceLevel, rating, address, website } = places[i];
+    const addressClass = address ? '' : 'undefined';
+    const phoneNumberClass = phoneNumber ? '' : 'undefined';
+    const ratingClass = rating ? '' : 'undefined';
+    const priceLevelClass = priceLevel ? '' : 'undefined';
+    const websiteClass = website ? '' : 'undefined';
+
+    placeCards.push(
+      `
+        <div class="col s12 m6">
+          <div class="card large">
+            <div class="card-image">
+              <img class="responsive-img" src="${photoUrl}" alt="${name}" loading="lazy">
+              <span class="card-title"><strong>${name}</strong></span>
+            </div>
+            <div class="card-fab">
+              <a class="btn-floating halfway-fab waves-effect waves-light blue">
+                <i class="material-icons">add</i>
+              </a>
+            </div>
+            <div class="card-content">
+              <p class="${addressClass}">${address}</p>
+              <p class="${phoneNumberClass}">${phoneNumber}</p>
+              <p class="${ratingClass}">Rating: ${rating}</p>
+              <p class="${priceLevelClass}">Price Level: ${priceLevel}</p>
+            </div>
+            <div class="${websiteClass} card-action">
+              <a href="${website}">Website</a>
+            </div>
+          </div>
+        </div>
+      `
+    )
+  }
+
+  placeCardsContainer.innerHTML = placeCards.join('');
+
+  let undefElements = placeCardsContainer.querySelectorAll('.undefined');
+  for(let i = 0; i < undefElements.length; i++) {
+    undefElements[i].remove();
+  }
+}
 
+async function getPlaceDetails(placeId) {
+  const detailsResponse = await fetch(
+    `https://cors-anywhere.herokuapp.com/https://maps.googleapis.com/maps/api/place/details/json?place_id=${placeId}&key=${GOOGLE_API_KEY}`
+  )
+  const { result } = await detailsResponse.json();
+  const { international_phone_number, name, photos, price_level, rating, vicinity, website } = result;
+  const photoUrl = await imageURLFromPhotos(photos);
+  
+  const placeDetails = {
+    phoneNumber: international_phone_number,
+    name: name,
+    photoUrl: photoUrl,
+    priceLevel: price_level,
+    rating: rating,
+    address: vicinity,
+    website: website
+  }
+
+  return placeDetails;
+}
+
+
+async function imageURLFromPhotos(photos) {
```

Comment:
This seems to be really similar to my `imageURLFromPhotoRef` function in `my-trips.js`. Maybe we can refactor this to be shared in `common.js`? Or is this a special case

Reply:
Yeah, I reimplemented your code for prototype's sake and I didn't want to accidentally break anything on your page, but I'll definitely refactor this code and maybe other common functions (find hotel or get place details) in a shared js file later on.

Reply:
It's come to my realization that our pages are extremely similar now, only difference being that the in one the user manually types out locations and while the other has the user select location suggestions.

Reply:
Yeah, I think the original direction was to enter a city and just have a bunch of locations pre-planned. I think you can still do that with rating info / popularity for the final version?

Reply:
For sure, some sorting based on rating and popularity as well as pre-added filters and that can be implemented.

Reply:
Actually, I remember now that I had meant to do this by allowing the user to select multiple filters at once, but Text Search only allows one type per call and one call is already almost prohibitively slow, taking up to 10 seconds to load results on the page. A version of the original plan can still be done by slimming down the list to only the top results based on rating and popularity however.